platforms/documentation/docs/src/snippets/bestPractices/avoidComputationInConfigurationPhase-do/groovy/build.gradle

```diff
@@ -0,0 +1,15 @@
+// tag::do-this[]
+abstract class MyTask extends DefaultTask {
+    @TaskAction
+    void run() {
+        heavyWork() // <1>
```

Comment:
💭 Trying to think as a novice here - `heavyWork()` is the _only_ work this task performs, which might be confusing and misinterpreted.  Perhaps it would be good to simulate both some necessary configuration time work and definitely execution time work via other outputs.  Then the benefits of moving `heavyWork`, which could be done in either phase, might contrast better and not unintentionally suggest moving _all_ work to execution time.